### PR TITLE
GOV-359: Add batch aggregate worker to bulk processor

### PR DESCRIPTION
**Pull request by arkadasfynarfin**
[GOV-359](https://mifosforge.jira.com/browse/GOV-359)

[GOV-359]: https://mifosforge.jira.com/browse/GOV-359?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by arkadasfynarfin** [comment]
@fynmanoj Please review this PR.
CC: @avikganguly01

**Comment by fynmanoj** [comment]
rebase and resolve conflicts

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/BatchStatusWorker.java [hunk 1]
```diff
@@ -1,52 +1,67 @@
...
+            logger.debug("Job '{}' started from process '{}' with key {}", job.getType(), job.getBpmnProcessId(), job.getKey());
 
-            Exchange exchange = new DefaultExchange(camelContext);
-            exchange.setProperty(BATCH_ID, variables.get(BATCH_ID));
-            exchange.setProperty(TENANT_ID, variables.get(TENANT_ID));
+            Map<String, Object> variables = job.getVariablesAsMap();
+            String batchId = (String) variables.get(BATCH_ID);
+            String tenantId = (String) variables.get(TENANT_ID);
+            BatchDTO batchDTOResponse = invokeBatchAggregationApi(batchId, tenantId);
+            long successRate = calculateSuccessPercentage(batchDTOResponse);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/BatchStatusWorker.java:31]: use float for percentage

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -29,13 +29,15 @@ zeebe:
...
-  contactpoint: "https://ops-bk.sandbox.mifos.io"
+#  contactpoint: "https://ops-bk.sandbox.mifos.io"
+  contactpoint: "http://localhost:5002"
   username: "mifos"
   password: "password"
   endpoints:
     auth: "/oauth/token"
     batch-summary: "/api/v1/batch"
     batch-transaction: "/api/v1/batch/transactions"
+    batch-aggregate: "/api/v1/batch/"
```
Comment (fynmanoj) [inline src/main/resources/application.yaml:40]: I would recommend you stick with the camelcase property definitions.

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: demo.csv [hunk 3]
```diff
@@ -0,0 +1,6 @@
+id,request_id,payment_mode,payer_identifier_type,payer_identifier,payee_identifier_type,payee_identifier,amount,currency,note
```
Comment (fynmanoj) [inline demo.csv:1]: remove this file

File: src/main/java/org/mifos/processor/bulk/API/BatchTransactions.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import javax.mail.Multipart;
+
+import java.io.IOException;
+
+import static org.mifos.processor.bulk.zeebe.ZeebeVariables.*;
+
+public interface BatchTransactions {
+
+    @PostMapping(value = "/batchtransactions", produces="application/json")
+    String batchTransactions(@RequestHeader(value = "X-CorrelationID") String requestId,
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/API/BatchTransactions.java:19]: please rebase, This is already added

File: src/main/java/org/mifos/processor/bulk/config/BudgetAccountConfig.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+@Getter
+@Setter
+@Component
+@ConfigurationProperties(prefix = "budget-account")
+public class BudgetAccountConfig {
+
+    private List<RegisteringInstitutionConfig> registeringInstitutions = new ArrayList<>();
+
+    public RegisteringInstitutionConfig getByRegisteringInstituteId(String id) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/config/BudgetAccountConfig.java:18]: please rebase

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Pull request closed without merging**